party_time/party_time.lua

```diff
@@ -0,0 +1,346 @@
+local os = require('os')
+local ui = require('ui')
+local sets = require("sets")
+local string = require('string')
+local command = require('command')
+local packets = require('packets')
+local settings = require('settings')
+local treasure = require('treasure')
+
+local defaults = {
+    ui = {
+        x = 145,
+        y = 440,
+        enabled = true,
+    },
+    auto = {
+        accept_invites = true,   -- also for auto sending invites upon party requests
+        decline_invites = false, -- also ignores party requests from players on blacklists
+    },
+    blacklist = {}, 
+    whitelist = {},
+}
+settings.settings_change:register(function(options)
+    options.blacklist = sets(options.blacklist)
+    options.whitelist = sets(options.whitelist)
+end)
+options = settings.load(defaults)
+options.blacklist = sets(options.blacklist)
+options.whitelist = sets(options.whitelist)
+
+local invite_dialog = {}
+local invite_dialog_state = {
+    title = 'Party Invite',
+    style = 'normal',
+    x = options.ui.x,
+    y = options.ui.y,
+    width = 179,
+    height = 96,
+    resizable = false,
+    moveable = true,
+    closable = true,
+}
+local unhandled_requests = {}
+local unhandled_invite = false
+
+-- Command Arg Types
+-- Create and Register types
+local arg_lookups = {
+    addremove = {
+        ['add'] = 'add',
+        ['a'] = 'add',
+        ['+'] = 'add',
+        ['remove'] = 'remove',
+        ['r'] = 'remove',
+        ['-'] = 'remove',
+    },
+    bool = {
+        ['yes'] = true,
+        ['y'] = true,
+        ['on'] = true,
+        ['no'] = false,
+        ['n'] = false,
+        ['off'] = false,
+    },
+}
+
+command.arg.register('name', '<name:string(%a+)>')
+
+command.arg.register_type('lookup_boolean', {
+    check = function(str)
```

Comment:
I would leave it as it is, though I think there would be merit to extracting that functionality into core. It's rather common to want to leave multiple options for certain commands, so having such a mechanism provided by default might be useful.